templates/project/build.rs

```diff
@@ -1,16 +1,59 @@
 use cargo_metadata::{CargoOpt, DependencyKind, MetadataCommand};
 use regex::Regex;
+use serde::{Deserialize, Serialize};
 use std::env;
 
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub struct Config {
+    pub cloud: Cloud,
+}
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub struct Cloud {
+    pub id: String,
+    pub secret: String,
+    pub app_address: String,
+}
+
 fn main() {
     if Regex::new(r"\w+-esp3?2?s?\d?-espidf")
         .unwrap()
         .is_match(&env::var("TARGET").unwrap())
     {
-        embuild::build::CfgArgs::output_propagated("MICRO_RDK").unwrap();
-        embuild::build::LinkArgs::output_propagated("MICRO_RDK").unwrap();
+        println!("cargo:rerun-if-changed=viam.json");
```

Comment:
Maybe move this line down to the block below where we are dealing with `viam.json`?